baselinemarkers/get_marker_genes_rnaseq.R

```diff
@@ -0,0 +1,363 @@
+
+# Identification of markers genes for baseline rna-seq experiments in Expression Atlas.
+
+
+required_packages <- c("MGFR", "xml2", "dplyr", "data.table") 
+
+# function to check and load a package, fail if not installed
+require_or_fail <- function(pkg) {
+  if (!requireNamespace(pkg, quietly = TRUE)) {
+    stop(paste("Package", pkg, "is required but not installed."), call. = FALSE)
+  }
+  suppressPackageStartupMessages(library(pkg, character.only = TRUE))
+}
+
+# check and load each
+for (pkg in required_packages) {
+  require_or_fail(pkg)
+}
+cat("All required packages are loaded successfully.\n")
+
+
+
+## parse arguments
+args <- commandArgs(trailingOnly = TRUE)
+
+if (length(args) != 6) {
+  stop("This script requires exactly 6 arguments: <config_xml> <expression_data.undecorated>  <expression_data>  <output_file> <specificity_score_cuttoff> <expression_cuttoff>")
```

Comment:
May be worth using`<EXP_ID>` to reduce number of arguments provided by calling script

`<EXP_ID> <specificity_score_cuttoff> <expression_cuttoff>`

`<config_xml> <expression_data.undecorated> <expression_data> <output_file>` can be predicted/generated using `<EXP_ID>`

Reply:
Also this can be simplified even more by using default for `SPECIFICITY_SCORE_CUTOFF` and `EXPRESSION_CUTOFF` and assign if provided